### PR TITLE
Add geoserver printing host customiization

### LINKED ISSUE #223: Feature: GeoServer print module configuration (allowed hosts)
## Feature Description

The GeoServer print module has a configuration file which contains among other things the list of the allowed hosts the module can connect to.

Since the 4.4 version, the default configuration has changed by the removal of the rule allowing to connect to any host. As a consequence, the use of the print module no longer works on a production environment (the public URL of the geonode is no longer allowed).

## Proposed Solution

The solution is to at least inject on the configuration file (geoserver/data/printing/config.yaml) the rule to work with the public URL of the GeoNode instance. If possible, it would be nice to be able to add other allowed hosts in the configuration file.

## Additional Information

Any additional information or context that may be helpful in implementing the feature.

### PR DESCRIPTION
**Pull request by AlexGacon**
## Description

This PR completes the configuration of the printing service of GeoServer to works with the URL of GeoNode and adds the possibility to define additional hosts in the values.yaml file.

## Type of Change

Please select the relevant option:

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Refactoring
- [ ] Other (please describe)

## Related Issue

If there is an existing issue related to this pull request, please reference it here.

closes #223 

## Checklist

Please ensure that your pull request meets the following requirements:

- The pull request is limited to one type (docs, feature, bug fix, etc.)
- The pull request is as small as possible. Consider opening multiple pull requests instead of one large one.
- The feature or bug fix has been discussed and documented in an issue beforehand.

## Additional Notes

Any additional information or context regarding the pull request can be provided here.

Thank you for creating this pull request

### CODE REVIEW

**Review by mwallschlaeger — CHANGES_REQUESTED** [review]
@AlexGacon for me its very unclear where all the configuration inside of the `charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml` files comes from. Are this default values from the GeoServer or did you gather them by yourself?

#edit: ok after reading the related issue I fully overview the problem. Maybe you could at least add a header to the yaml file describing where the content is comming from and that its served or controlled by geonode-k8s helm chart. So people manually straving inside the geoserver container will know what they are looking at.

File: charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+      - 70000000.0
+      - 140000000.0
+      - 280000000.0
+
+    outputFilename: 'GeoNode-Map-${date}.pdf'
+    #== ms2 ===== 
+    disableScaleLocking: true
+    #======
+    brokenUrlPlaceholder: 'default'
+    # brokenUrlPlaceholder: 'throw'
```
[the remark this answers is not in the document]
Comment (AlexGacon) [reply charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml]: Done

File: charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml [hunk 2]
```diff
@@ -0,0 +1,1291 @@
...
+
+    #proxyBaseUrl: http://geoserver:8080/geoserver/pdf
```
[the remark this answers is not in the document]
Comment (AlexGacon) [reply charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml]: Done

File: charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml [hunk 3]
```diff
@@ -0,0 +1,1291 @@
...
+    connectionTimeout: 2000
+    socketTimeout: 2000
+
+    #===========================================================================
+    # the list of allowed ips
+    #===========================================================================
```
[the remark this answers is not in the document]
Comment (AlexGacon) [reply charts/geonode/templates/geoserver/geoserver-printing-config-yaml.yaml:65]: Documentation added and I have moved the custom part at the end of the list, with additional comments.

**Comment by AlexGacon** [comment]
@mwallschlaeger is the removal of notifier.xml ok for you?

**Comment by mwallschlaeger** [comment]
@AlexGacon yes removing of the xml is fine, as its served via geoserver docker image

**Review by mwallschlaeger — APPROVED** [review]
Looks fine to me now in total.

**Pull request merged**